### PR TITLE
docs and simplification of cmd args

### PR DESCRIPTION
**Pull request by arendu**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Jenkins CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

There's no need to comment `jenkins` on the PR to trigger Jenkins CI.
The GitHub Actions CI will run automatically when the PR is opened.
To run CI on an untrusted fork, a NeMo user with write access must click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [x] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by arendu** [comment]
jenkins

File: docs/source/nlp/information_retrieval.rst [hunk 1]
```diff
@@ -102,3 +102,107 @@ Then you can fine-tune the sentence-BERT model using the following script:
     exp_manager.wandb_logger_kwargs.name=${NAME} \
     exp_manager.wandb_logger_kwargs.project=${PROJECT}
     
+GPT Embedding Models
+=====================
+
+Recent work has also shown that it is possible to use Decoder-Only (GPT Style) models to train embedding models.
+`Improving Text Embeddings with
+Large Language Models <https://arxiv.org/pdf/2401.00368.pdf>`__ is one such recent papers which served as inspritation to implement Decoder-only embdding training in Nemo.
```
Comment (JRD971000) [inline docs/source/nlp/information_retrieval.rst:110]: embdding -> embedding 
inspritation - > inspiration

File: docs/source/nlp/information_retrieval.rst [hunk 2]
```diff
@@ -102,3 +102,107 @@ Then you can fine-tune the sentence-BERT model using the following script:
...
+
+During training, the GPT Embedding model employs LoRA (by default) to learn embeddings for the queries and documents, such that similarity of the ``query``-to-``pos_doc`` are maximized while simultaneously minimizing ``query``-to-``neg_doc`` similarity. LoRA allows us to fine-tune large LLMs such as Mistral 7B model with a relatively small number of training parameters.
+
+An example command to launch a training job is
+
+.. code-block:: console
+
+ python3 /NeMo/examples/nlp/information_retrieval/megatron_gpt_embedding_finetuning.py \
+    exp_manager.exp_dir="PATH_TO_SAVE_LORA_WEIGHTS" \
+    model.global_batch_size=4 \                         # exact choice for global batch size is data dependent typical values are in the rage of 32 to 128.
```
Comment (JRD971000) [inline docs/source/nlp/information_retrieval.rst:132]: rage -> range

File: docs/source/nlp/information_retrieval.rst [hunk 3]
```diff
@@ -102,3 +102,107 @@ Then you can fine-tune the sentence-BERT model using the following script:
...
+{"pos_doc": "Exceptional examples ... Warsaw.", "doc_id": "d1084"}
+
+Once again, we show 3 examples form each file. Typically the ``test_docs.jsonl`` will contain more items than queries in the ``test_queries.jsonl``.
+
+The inference command will result in two folders 
+
+* ``PATH_TO_SAVE_EMBEDDINGS/consumed_samplesX/test_queries`` 
+* ``PATH_TO_SAVE_EMBEDDINGS/consumed_samplesX/test_docs``
+
+The ``X`` in the folder ``consumed_samplesX`` is a number denoted number of batches consumed, this is not crucial at test time, but it is useful in training which we will see in the next section. First, lets take a look at the ``test_queries``.
```
Comment (JRD971000) [inline docs/source/nlp/information_retrieval.rst:184]: lets - > let's

File: docs/source/nlp/information_retrieval.rst [hunk 4]
```diff
@@ -102,3 +102,107 @@ Then you can fine-tune the sentence-BERT model using the following script:
...
+ $> ls PATH_TO_SAVE_EMBEDDINGS/consumed_samplesX/test_queries
+ query.ids  query.npy
+ $>head -n3 PATH_TO_SAVE_EMBEDDINGS/consumed_samplesX/test_queries/query.ids 
+ q11600
+ q5831
+ q3037
+
+``query.npy`` is a numpy pickled array containing rows of query embeddings and the ``query.ids`` text file list the id of each embedding in the same order.
+
+Similarly lets look into the ``test_docs`` folder
```
Comment (JRD971000) [inline docs/source/nlp/information_retrieval.rst:197]: lets -> let's

**Comment by JRD971000** [comment]
Great doc @arendu , just a few typos to fix. Btw, I saw no Jenkins test, is cicd-main.yml the replacement?

**Review by JRD971000 — APPROVED** [review]

**Review by JRD971000 — DISMISSED** [review]

**Review by JRD971000 — APPROVED** [review]

**Pull request merged**